extensions/2.0/Khronos/KHR_avatar_expression_procedural/README.md

```diff
@@ -0,0 +1,118 @@
+Dependent on: `KHR_avatar`
+Works alongside: `KHR_avatar_expression_mapping`, `KHR_avatar_expression_joint`, `KHR_avatar_expression_morphtargets`.
+
+## Overview
+
+The `KHR_avatar_expression_procedural` extension introduces a metadata layer that defines which expressions are expected to be procedurally generated or controlled at runtime. This includes expressions such as:
+
+- **Timed blinking**
+- **Eye darting**
+- **Idle mouth movement**
+- **Microexpressions**
+- **Randomized twitches or gestures**
+
+By labeling expressions as procedural, this extension enables runtime systems (e.g., animation controllers, live-tracking systems, or AI avatars) to drive those expressions without conflict with animation data or user inputs.
+
+This metadata is **descriptive only**: it does not contain animation or behavior logic itself. Instead, it signals to tools and engines that the following expressions are _expected to be dynamically generated_.
+
+## Use Cases
+
+- Define which expressions are not statically animated but generated procedurally
+- Avoid conflict between procedural and baked expression systems
+- Inform rig exporters or runtime systems which expressions are reserved for real-time generation
+- Enable consistent behavior across platforms
+
+## Schema
+
+```json
+{
+  "extensions": {
+    "KHR_avatar_expression_procedural": {
+      "proceduralExpressions": [
+        {
+          "expression": "blinkLeft",
+          "mode": "timed"
+        },
+        {
+          "expression": "mouthTwitch",
+          "mode": "random"
+        }
+      ]
+    }
+  }
+}
+```
+
+### Properties
+
+| Property                | Type   | Description                                                                             |
+| ----------------------- | ------ | --------------------------------------------------------------------------------------- |
+| `proceduralExpressions` | array  | List of expression metadata entries that are intended to be procedurally controlled     |
+| `expression`            | string | Expression name as defined in the expression mapping vocabulary                         |
+| `mode`                  | string | Enum: `"timed"`, `"random"`, `"live"`, `"scripted"` — describes the procedural strategy |
+
+## Procedural Modes
+
+| Mode       | Meaning                                                                        |
+| ---------- | ------------------------------------------------------------------------------ |
+| `timed`    | The expression occurs in regular intervals (e.g., blinking every 5s)           |
+| `random`   | The expression fires with stochastic timing or weights                         |
+| `live`     | Driven by live-captured inputs (e.g., webcam, eye tracking)                    |
+| `scripted` | Driven by external dialogue, AI logic, or gameplay sequences                   |
+| `mixed`    | Combines two or more procedural sources, such as live tracking + scripted cues |
```

Comment:
I prefer not to provide this hint from the avatar side, especially `timed` vs. `random`. I want to fully delegate to the app the decision on how to control blinks, since there are many ways to control blinks, considering performance and expressiveness.
For `live` vs `mixed`, it depends on the application's availability and the user's preference. Even if the app can provide eye blinking via live tracking procedurally, the user might not be content with the quality of the expression.